### PR TITLE
Fix type of SSLContext for some static type checkers

### PR DESCRIPTION
**Pull request by AlanBogarin**
## What do these changes do?

Fix partially unknown bug in all references to SSLContext

## Are there changes in behavior for the user?

No, it's only for type checkers

## Is it a substantial burden for the maintainers to support this?

No, it only requires knowledge of run-time and type checking.

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `CHANGES/` folder
  * name it `<issue_or_pr_num>.<type>.rst` (e.g. `588.bugfix.rst`)
  * if you don't have an issue number, change it to the pull request
    number after creating the PR
    * `.bugfix`: A bug fix for something the maintainers deemed an
      improper undesired behavior that got corrected to match
      pre-agreed expectations.
    * `.feature`: A new behavior, public APIs. That sort of stuff.
    * `.deprecation`: A declaration of future API removals and breaking
      changes in behavior.
    * `.breaking`: When something public is removed in a breaking way.
      Could be deprecated in an earlier release.
    * `.doc`: Notable updates to the documentation structure or build
      process.
    * `.packaging`: Notes for downstreams about unobvious side effects
      and tooling. Changes in the test invocation considerations and
      runtime assumptions.
    * `.contrib`: Stuff that affects the contributor experience. e.g.
      Running tests, building the docs, setting up the development
      environment.
    * `.misc`: Changes that are hard to assign to any of the above
      categories.
  * Make sure to use full sentences with correct case and punctuation,
    for example:
    ```rst
    Fixed issue with non-ascii contents in doctest text files
    -- by :user:`contributor-gh-handle`.
    ```

    Use the past tense or the present tense a non-imperative mood,
    referring to what's changed compared to the last released version
    of this project.

### CODE REVIEW

**Comment by Dreamsorcerer** [comment]
We've been using this without issue for quite some time. You'll need to provide a reproducer demonstrating the bug you're claiming to fix.

**Comment by webknjaz** [comment]
I don't even know if the try/except is needed these days. I don't know of a Python distribution being shipped without the `ssl` module by default post Python 2.7.9. Someone could disable it when compiling CPython but I don't know that we want to support such custom builds anyway.

**Comment by Dreamsorcerer** [comment]
Yep, maybe it's time to remove it then.

**Comment by asvetlov** [comment]
Python itself support the build without ssl.
Asyncio has such conditional imports also.
I think we should keep it until CPython decide to make ssl mandatory (as it happened for threading module).

**Review by asvetlov — APPROVED** [review]
OK, makes sense.
Looks like mypy has a different logic  than pyright, that's why we never encountered the error in out liter runs.

**Comment by Dreamsorcerer** [comment]
It's a bit awkward, but looks like we'll have to go with it.

Please put the no cover comments back in first though, as you can see from the codecov comment above, this has caused a regression in coverage.

File: CHANGES/10099.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed unknown `SSLContext` symbol for type checkers
```
Comment (Dreamsorcerer) [inline CHANGES/10099.bugfix.rst]:
```suggestion
Fixed type of ``SSLContext`` for some static type checkers (e.g. pyright).
```

Comment (bdraco) [inline CHANGES/10099.bugfix.rst]:
```suggestion
Fixed unknown `SSLContext` symbol for type checkers -- by :user:`AlanBogarin`.
```

Comment (bdraco) [reply CHANGES/10099.bugfix.rst]: It looks like this suggestion was marked as resolved without any context or commentary.

**Review by bdraco — CHANGES_REQUESTED** [review]
Please see https://github.com/aio-libs/aiohttp/pull/10099#discussion_r1871966506

**Comment by bdraco** [comment]
Usually we add credit to the end change note in the form. It's not clear if you are explicitly wanting to not be credited or its on oversight. 

```
 -- by :user:`AlanBogarin`.
```

File: aiohttp/client_exceptions.py [hunk 2]
```diff
@@ -7,13 +7,17 @@
...
 
     SSLContext = ssl.SSLContext
-except ImportError:  # pragma: no cover
-    ssl = SSLContext = None  # type: ignore[assignment]
+else:
+    try:
+        import ssl
 
+        SSLContext = ssl.SSLContext
+    except ImportError:
```
Comment (Dreamsorcerer) [inline aiohttp/client_exceptions.py]: Not sure if you missed my other comment, but still a coverage regression:
https://github.com/aio-libs/aiohttp/pull/10099#issuecomment-2520352679

Comment (AlanBogarin) [reply aiohttp/client_exceptions.py]: I did (`# pragma: no cover`), Wouldn't it be better to remove the try/except block?

**Comment by AlanBogarin** [comment]
What does `CI/Linter (pull_request)` do?

**Comment by bdraco** [comment]
> What does `CI/Linter (pull_request)` do?

https://github.com/aio-libs/aiohttp/actions/runs/12192823875/job/34014087992?pr=10099

`[towncrier-fragments]:194: : Spell check: pyright:  for some static type checkers (e.g. pyright)..`

File: aiohttp/client_exceptions.py (hunk 2 above)
Comment (Dreamsorcerer) [reply aiohttp/client_exceptions.py]: Well, yes, that's what we suggested earlier. We don't need the TYPE_CHECKING either then. asvetlov is keen to keep it for now though (https://github.com/aio-libs/aiohttp/pull/10099#issuecomment-2517971410). Though I can't imagine anyone using a version of Python without SSL and wanting to use aiohttp on it..

**Comment by Dreamsorcerer** [comment]
That was my suggestion, have adding to spelling list. Coverage is good now too.

**Review by Dreamsorcerer — APPROVED** [review]

**Review by bdraco — APPROVED** [review]

**Pull request merged**